src/Selenium2Driver.php

```diff
@@ -758,170 +695,135 @@ public function selectOption($xpath, $value, $multiple = false)
      */
     public function isSelected($xpath)
     {
-        return $this->findElement($xpath)->selected();
+        $element = $this->findElement($xpath);
+        return $element->isSelected();
     }
 
     /**
      * {@inheritdoc}
      */
     public function click($xpath)
     {
-        $this->clickOnElement($this->findElement($xpath));
+        $element = $this->findElement($xpath);
+        $this->clickOnElement($element);
     }
 
-    private function clickOnElement(Element $element)
+    private function clickOnElement(WebDriverElement $element)
     {
-        try {
-            // Move the mouse to the element as Selenium does not allow clicking on an element which is outside the viewport
-            $this->wdSession->moveto(array('element' => $element->getID()));
-        } catch (UnknownCommand $e) {
-            // If the Webdriver implementation does not support moveto (which is not part of the W3C WebDriver spec), proceed to the click
-        }
-
-        $element->click();
+        // Move the mouse to the element as Selenium does not allow clicking on an element which is outside the viewport
+        $this->webDriver->action()->click($element)->perform();
     }
 
     /**
      * {@inheritdoc}
      */
     public function doubleClick($xpath)
     {
-        $this->mouseOver($xpath);
-        $this->wdSession->doubleclick();
+        $element = $this->findElement($xpath);
+        $this->webDriver->action()->doubleClick($element)->perform();
     }
 
     /**
      * {@inheritdoc}
      */
     public function rightClick($xpath)
     {
-        $this->mouseOver($xpath);
-        $this->wdSession->click(array('button' => 2));
+        $element = $this->findElement($xpath);
+        $this->webDriver->action()->contextClick($element)->perform();
     }
 
     /**
      * {@inheritdoc}
      */
     public function attachFile($xpath, $path)
     {
+        /** @var RemoteWebElement $element */
```

Comment:
this phpdoc should be useless. The return type should already be documented properly.

Reply:
updated